test/unit/error.spec.ts

```diff
@@ -24,4 +24,45 @@ export class LuaErrorTests {
             );
         }).toThrowError(TranspileError, "Invalid throw expression, only strings can be thrown.");
     }
+
+    @TestCase(1, "ad")
+    @TestCase(2, "bcd")
+    @TestCase(3, "e")
+    @TestCase(4, "g")
+    @TestCase(5, "h")
+    @TestCase(6, "i")
+    @Test("re-throw")
+    public reThrow(count: number, expected: any): void {
+        const source =
+            `declare namespace string { export function match(s: string, p: string): string | null; }
+            let i = ${count};
+            function foo(s: string) { if (--i <= 0) { throw s; } }
```

Comment:
This mechanism actually makes it really hard to see what this test does, I would suggest splitting this up into multiple tests or maybe even just doing a `if (count == 1) { throw "a" }` instead of `foo(`

Reply:
Yes, it's pretty convoluted. I'll come up with something better.